doozer/doozerlib/backend/konflux_image_builder.py

```diff
@@ -0,0 +1,319 @@
+            "spec": {
+                "displayName": display_name,
+                "appModelRepository": {"url": ""},
+                "gitOpsRepository": {"url": ""},
+            }
+        }
+        return obj
+
+    @staticmethod
+    async def _get_pipelinerun_api(dyn_client: DynamicClient):
+        return await exectools.to_thread(
+            dyn_client.resources.get,
+            api_version="tekton.dev/v1",
+            kind="PipelineRun",
+        )
+
+    @staticmethod
+    def _new_component(name: str, application: str, component_name: str,
+                       image_repo: Optional[str], source_url: Optional[str], revision: Optional[str]) -> dict:
+        obj = {
+            "apiVersion": "appstudio.redhat.com/v1alpha1",
+            "kind": "Component",
+            "metadata": {
+                "name": name,
+                "annotations": {
+                    "build.appstudio.openshift.io/pipeline": '{"name":"docker-build-multi-platform-oci-ta","bundle":"latest"}',
+                    # "build.appstudio.openshift.io/request": "configure-pac",
+                    "mintmaker.appstudio.redhat.com/disabled": "true",  # https://gitlab.cee.redhat.com/konflux/docs/users/-/blob/main/topics/mintmaker/user.md#offboarding-a-repository
+                }
+            },
+            "spec": {
+                "application": application,
+                "componentName": component_name,
+                "source": {
+                    "git": {
+                        "url": source_url,
+                        "revision": revision,
+                    }
+                }
+            }
+        }
+        if image_repo:
+            obj["spec"]["containerImage"] = image_repo
+        if source_url:
+            obj["spec"].setdefault("source", {}).setdefault("git", {})["url"] = source_url
+        if revision:
+            obj["spec"].setdefault("source", {}).setdefault("git", {})["revision"] = revision
+        return obj
+
+    @staticmethod
+    def _new_pipelinerun(generate_name: str, application_name: str, component_name: str,
+                         git_url: str, commit_sha: str, target_branch: str, output_image: str,
+                         build_platforms: Sequence[str], git_auth_secret: str = "pipelines-as-code-secret") -> dict:
+        https_url = art_util.convert_remote_git_to_https(git_url)
+        template_content = files("doozerlib").joinpath("backend").joinpath("konflux_image_build_pipelinerun.yaml").read_text()
+        template = jinja2.Template(template_content, autoescape=True)
+        rendered = template.render({
+            "source_url": https_url,
+            "revision": commit_sha,
+            "target_branch": target_branch,
+            "git_auth_secret": git_auth_secret,
+
+        })
+        obj = yaml.load(rendered)
+        del obj["metadata"]["name"]
+        del obj["metadata"]["namespace"]
+        del obj["metadata"]["annotations"]["pipelinesascode.tekton.dev/on-cel-expression"]
```

Comment:
curious what's the reason behind this / or why do we have this in yaml in the first place if we don't want them?

Reply:
In the future the PipelineRun template file (https://github.com/openshift-eng/art-tools/pull/859/files#diff-f1894aed6cfd6dbe45cd9701265dbf035239886d026b4dca0c40865f5e8511fe) will be loaded from an external repo, which includes some fields that are specific to a single image. We'd better to remove them.

Reply:
I'll add a comment here